doc/provisioning.md

```diff
@@ -1,19 +1,28 @@
-What We Run, and Why
-====================
-
-Expressed as compactly as possible, the Provisioner is responsible for making
-sure that non-Dead machine entities in state have agents running on live
-instances; and for making sure that Dead machines, and stray instances, are
-removed and cleaned up.
-
-However, the choice of exactly what we deploy involves some subtleties. At the
-Provisioner level, it's simple: the base and the constraints we pass to the
-Environ.StartInstance come from the machine entity. But how did they get there?
-
-Bases
------
-
-Individual charms are released for different possible target base; juju
+# Provisioning
+
+Provisioning in Juju is performed by two separate workers:
+- The [compute provisioner](https://github.com/juju/juju/blob/main/internal/worker/computeprovisioner/doc.go),
+which is responsible for provisioning machine instances on the underlying
+provider.
+- The [container provisioner](https://github.com/juju/juju/blob/main/internal/worker/containerprovisioner/doc.go),
+which is responsible for creating containers managed by the container
+broker on the underlying machine.
+
+Juju splits the resource creation in two stages. When a new machine
```

Comment:
**suggestion:** This is a very long sentence:

> When a new machine is being created (either by the user explictly or because a new unit
is deployed) the machine is first stored in the database, and then
one of the provisioners (compute or container) workers will pick it
up and start the provisioning process.

I think that side comments (in brackets), should be removed or inserted in the form of a footnote[^1]

[^1]: Maybe we should check that sphinx support it, in order to stay compatible with ReadTheDocs.

Reply:
We haven't been doing that for `docs/` so I'll keep it like that. Happy to change later if this becomes a convention.

Reply:
Ok no issue, I will try it one day when i will write a documentation, to see if it make people angry or if it is ok.

In the later case, it will become a convention :D

Reply:
This will be affected by how we want to view these docs. My initial thought was that we might publish them on RTD but as of yesterday I'm beginning to think that'll just create busywork and potentially confuse things with links etc. Dev docs are for devs, so they should be published in whichever way suits devs best.